sn_networking/src/record_store.rs

```diff
@@ -1461,120 +1460,140 @@ mod tests {
-                ) {
-                    Ok(peers_in_replicate_range) => {
-                        let peers_in_replicate_range: Vec<PeerId> = peers_in_replicate_range
-                            .iter()
-                            .map(|peer_id| **peer_id)
-                            .collect();
-                        let peers_in_close: Vec<PeerId> = match sort_peers_by_key(
-                            &peers_in_replicate_range,
-                            &address.as_kbucket_key(),
-                            close_group_majority(),
-                        ) {
-                            Ok(peers_in_close) => {
-                                peers_in_close.iter().map(|peer_id| **peer_id).collect()
-                            }
-                            Err(err) => {
-                                panic!("Can't find close range of {name:?} with error {err:?}")
-                            }
-                        };
-
-                        let payee = pick_cheapest_payee(&peers_in_close, &peers);
-
-                        for peer in peers_in_replicate_range.iter() {
-                            let (close_records_stored, nanos_earnt, received_payment_count) =
-                                peers.entry(*peer).or_insert((0, 0, 0));
-                            if *peer == payee {
-                                let cost = calculate_cost_for_records(&QuotingMetrics {
-                                    close_records_stored: *close_records_stored,
-                                    max_records: MAX_RECORDS_COUNT,
-                                    received_payment_count: *received_payment_count,
-                                    live_time: 0,
-                                });
-                                *nanos_earnt += cost;
-                                *received_payment_count += 1;
-                            }
-                            *close_records_stored += 1;
+            let chunk_results: Vec<_> = (0..num_of_chunks_per_itr)
+                .into_par_iter()
+                .map(|_| {
+                    let name = xor_name::rand::random();
+                    let chunk_address = NetworkAddress::from_chunk_address(ChunkAddress::new(name));
+
+                    let chunk_distance_to_sorting = sorting_target_address.distance(&chunk_address);
+                    // Binary search to find the insertion point for the chunk
+                    let partition_point = peers.partition_point(|peer| {
+                        sorting_target_address.distance(&peer.address) < chunk_distance_to_sorting
+                    });
+
+                    // Collect close_group_size closest peers
+                    let mut close_group = Vec::with_capacity(close_group_size);
+                    let mut left = partition_point;
+                    let mut right = partition_point;
+
+                    while close_group.len() < close_group_size && (left > 0 || right < peers_len) {
```

Comment:
the `partition_point` returns the first index that not satisfy the judgement

using this sorted list against a random target, to figure out closest peers against another target,
the current code block has following glitches:
1, the `left` side shall be preferred, as their `distance interval` is smaller than the `right` side peers
2, the distance from different observer will be different, i.e. the peers around the partition_point not guaranteered to always being the closest peers to the new target
3, the while loop will terminate earlier when the partition_point  falls into the beginning ([0,1,2]) or end ([len-2, len-1, len]) of the entire peers list.

for the point 1 and 2, the `glitch` might be acceptable, as the test is just a `sim`